tests/integration/kubernetes/k8s-hostname.bats

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2024 Ant Group
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+
+	pod_name="test-pod-hostname"
+	get_pod_config_dir
+
+	yaml_file="${pod_config_dir}/pod-hostname.yaml"
+	add_allow_all_policy_to_yaml "${yaml_file}"
+
+    expected_name=$pod_name
+}
+
+@test "Validate Pod hostname" {
+	# Create pod
+	kubectl apply -f "${yaml_file}"
+
+	# Validate pod Ready 
+	kubectl wait --for=condition=Ready --timeout=$timeout pod $pod_name
+
+	# Validate the pod hostname
+	exec_cmd="hostname"
```

Comment:
Could specify this command directly in the pod spec and just use `kubectl logs` in this test?

Reply:
I tend to agree with @sprt. Running `sleep infinitity` in the container and then exec'ing into it is bit convoluted. Let's just run `hostname` in the container and check the output.

Reply:
Yeah, good suggestion, Thx both of you two.